packages/gatsby-plugin-netlify-cms/src/gatsby-node.js

```diff
@@ -56,115 +56,122 @@ exports.onCreateWebpackConfig = (
-            },
+          /**
+           * Provide a custom message for Netlify CMS compilation success.
+           */
+          stage === `develop` &&
+            new FriendlyErrorsPlugin({
+              clearConsole: false,
+              compilationSuccessInfo: {
+                messages: [
+                  `Netlify CMS is running at ${
+                    program.ssl ? `https` : `http`
+                  }://${program.host}:${program.port}/${publicPathClean}/`,
+                ],
+              },
+            }),
+
+          /**
+           * Use a simple filename with no hash so we can access from source by
+           * path.
+           */
+          new MiniCssExtractPlugin({
+            filename: `[name].css`,
           }),
 
-        /**
-         * Use a simple filename with no hash so we can access from source by
-         * path.
-         */
-        new MiniCssExtractPlugin({
-          filename: `[name].css`,
-        }),
+          /**
+           * Auto generate CMS index.html page.
+           */
+          new HtmlWebpackPlugin({
+            title: htmlTitle,
+            chunks: [`cms`],
+            excludeAssets: [/cms.css/],
+          }),
 
-        /**
-         * Auto generate CMS index.html page.
-         */
-        new HtmlWebpackPlugin({
-          title: htmlTitle,
-          chunks: [`cms`],
-          excludeAssets: [/cms.css/],
-        }),
+          /**
+           * Exclude CSS from index.html, as any imported styles are assumed to be
+           * targeting the editor preview pane. Uses `excludeAssets` option from
+           * `HtmlWebpackPlugin` config.
+           */
+          new HtmlWebpackExcludeAssetsPlugin(),
 
-        /**
-         * Exclude CSS from index.html, as any imported styles are assumed to be
-         * targeting the editor preview pane. Uses `excludeAssets` option from
-         * `HtmlWebpackPlugin` config.
-         */
-        new HtmlWebpackExcludeAssetsPlugin(),
+          /**
+           * Pass in needed Gatsby config values.
+           */
+          new webpack.DefinePlugin({
+            __PATH__PREFIX__: pathPrefix,
+            CMS_PUBLIC_PATH: JSON.stringify(publicPath),
+          }),
+        ].filter(p => p),
 
         /**
-         * Pass in needed Gatsby config values.
+         * Remove common chunks style optimizations from Gatsby's default
+         * config, they cause issues for our pre-bundled code.
          */
-        new webpack.DefinePlugin({
-          __PATH__PREFIX__: pathPrefix,
-          CMS_PUBLIC_PATH: JSON.stringify(publicPath),
-        }),
-      ].filter(p => p),
-
-      /**
-       * Remove mode and common chunks style optimizations from Gatsby's default
-       * config, they cause issues for our pre-bundled code.
-       */
-      mode: `none`,
-      optimization: {},
-      devtool: stage === `develop` ? `cheap-module-source-map` : `source-map`,
-    }
+        mode: stage === `develop` ? `development` : `production`,
```

Comment:
We should be explicit here, since this build is not really tied to the Gatsby build. Remember, this is an outside bundle separate from the Gatsby build. It has it's own webpack entry and output.

Reply:
I agree that it's not tied to gatsby but all gatsby config is used at the top so it's probably better to ride with gatsby on this or at least check NODE_ENV instead of stage :)

Reply:
but let's keep it for now as it was there before